### ISSUE TITLE
wr.db.read_sql_query tends to return object datatype rather than float or decimal on Redshift data

### ISSUE THREAD

**Issue by Eric-He-98** [awslabs/aws-data-wrangler#431]
**Describe the bug**
My team currently uses an internal class to connect to and read data from Redshift. The SQL read functionality wraps around the `pandas.read_sql` function. It's been my experience that pandas read_sql is better at inferring or perpetuating data types than the wr.db.read_sql_query function; any idea how this discrepancy might be closed? 

**To Reproduce**
Steps to reproduce the behavior. Also add details about Python and Wrangler's version and how the library was installed.
pandas==1.1.3
wranger==1.9.6

I received this data as the output of con.execute().fetchall() which is called by the `read_sql_query` function:
```
raw = [(30002637914, datetime.date(2019, 4, 21), Decimal('353.6900000000000000')),
 (30002815771, datetime.date(2019, 12, 2), Decimal('435.0100000000000000')),
 (30002193298, datetime.date(2019, 1, 30), Decimal('83.7400000000000000')),
 (30001592665, datetime.date(2018, 9, 4), Decimal('160.2000000000000000')),
 (30002292389, datetime.date(2018, 11, 12), Decimal('178.0800000000000000')),
 (30002067517, datetime.date(2018, 9, 20), Decimal('86.2500000000000000')),
 (30002467555, datetime.date(2019, 1, 25), Decimal('185.7054577000000000')),
 (30002949181, datetime.date(2019, 10, 19), Decimal('97.0300000000000000')),
 (30003014342, datetime.date(2019, 11, 25), Decimal('67.0500000000000000')),
 (30002385710, datetime.date(2019, 6, 17), Decimal('119.6900000000000000'))]
```

Feeding into the `_records2df` helper function gets me the following:
```
df = wr.db._records2df(records=raw,
                  cols_names=['user_id', 'date', 'sum'],
                  index=None)

df['sum']
>>> 0    353.6900000000000000
1    435.0100000000000000
2     83.7400000000000000
3    160.2000000000000000
4    178.0800000000000000
5     86.2500000000000000
6    185.7054577000000000
7     97.0300000000000000
8     67.0500000000000000
9    119.6900000000000000
Name: sum, dtype: object
```

Using pd.read_sql to fetch the data from Redshift, this column would return as float64 for me. It would be better if the decimal type could be returned as float type. I would prefer to have this done automatically but even asking the read_sql_query function to cast it to float or float64 throws a bug (possibly a pyarrow internal error):

```
df = wr.db._records2df(records=a,
                  cols_names=['user_id', 'date', 'sum'],
                  index=None,
                  dtype={'sum':'float'})

ArrowInvalid: Could not convert 353.6900000000000000 with type decimal.Decimal: tried to convert to float32

df = wr.db._records2df(records=a,
                  cols_names=['user_id', 'date', 'sum'],
                  index=None,
                  dtype={'sum':'float64'})

ArrowInvalid: Could not convert 353.6900000000000000 with type decimal.Decimal: tried to convert to double
```

**Comment by igorborgest** [comment]
Hi @Eric-He-98 !

I think that the object/decimal type is the right default in this case because it will not generate any loss. But if you prefer to go with `float`, we just implemented the possibility to cast it through the `dtype` argument (Check the commit above for more details).

Do you mind to install it directly from our repository and give some feedback before the official publishment?
The idea is to publish it in the version `1.10.0` this weekend.

```sh
pip install git+https://github.com/awslabs/aws-data-wrangler.git --no-use-pep517
```

**Comment by Eric-He-98** [comment]
looks like it works for me whether or not I use `safe=True` or `safe=False`. Thank you very much!

**Comment by igorborgest** [comment]
Released on [1.10.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.10.0)
